src/Take.Elephant.Azure/AzureEventHubReceiverQueue.cs

```diff
@@ -1,148 +1,144 @@
-        private readonly TimeSpan _maxDequeueRetryDelay;
         private readonly ISerializer<T> _serializer;
-        private readonly EventHubClient _eventHubClient;
-        private PartitionReceiver _receiver;
-        private readonly SemaphoreSlim _openSemaphore;
+        private readonly EventHubConsumerClient _consumer;
+        private readonly SemaphoreSlim _consumerStartSemaphore;
+        private readonly Channel<string> _channel;
+        private readonly CancellationTokenSource _cts;
+        private Task _consumerTask;
 
-        public AzureEventHubReceiverQueue(
-            string eventHubName,
-            string eventHubConnectionString,
-            ISerializer<T> serializer,
-            string consumerGroupName,
-            string partitionId, 
-            EventPosition eventPosition, 
-            ReceiverOptions receiverOptions = null,
-            int minDequeueRetryDelay = 250,
-            int maxDequeueRetryDelay = 30000)
+        public AzureEventHubReceiverQueue(string connectionString, string topic, string consumerGroup, ISerializer<T> serializer)
+            : this(new EventHubConsumerClient(consumerGroup, connectionString, topic), serializer)
         {
-            Guard.Argument(eventHubName).NotNull().NotEmpty();
-            Guard.Argument(eventHubConnectionString).NotNull().NotEmpty();
-            _consumerGroupName = Guard.Argument(consumerGroupName).NotNull().Value;
-            _eventPosition = Guard.Argument(eventPosition).NotNull().Value;
-            _partitionId = Guard.Argument(partitionId).NotNull().Value;
-            _receiverOptions = receiverOptions;
-            Guard.Argument(minDequeueRetryDelay).Max(maxDequeueRetryDelay);
-            _minDequeueRetryDelay = TimeSpan.FromMilliseconds(minDequeueRetryDelay);
-            _maxDequeueRetryDelay = TimeSpan.FromMilliseconds(maxDequeueRetryDelay);
-            _serializer = Guard.Argument(serializer).NotNull().Value;
-            
-            _eventHubClient = EventHubClient.CreateFromConnectionString(
-                new EventHubsConnectionStringBuilder(eventHubConnectionString)
-                {
-                    EntityPath = eventHubName
-                }.ToString());
-            _openSemaphore = new SemaphoreSlim(1);
         }
-        
-        public async Task<T> DequeueOrDefaultAsync(CancellationToken cancellationToken = default)
+
+        public AzureEventHubReceiverQueue(EventHubConsumerClient consumer, ISerializer<T> serializer)
         {
-            await OpenIfRequiredAsync(cancellationToken);
-            var eventDatas = await _receiver.ReceiveAsync(1, _minDequeueRetryDelay);
-            var eventData = eventDatas?.FirstOrDefault();
-            return CreateItem(eventData);
+            _consumer = consumer;
+            _serializer = serializer;
+            _consumerStartSemaphore = new SemaphoreSlim(1, 1);
+            _channel = Channel.CreateBounded<string>(1);
```

Comment:
Since we have a single writer here, we should create a `BoundedChannelOptions` and set `SingleWriter` on it